### PR TITLE
Store: Settings: Notices: Add redux

### PR DESCRIPTION
**Pull request by allendav**
Partially addresses #20623 

Adds selectors to extract store notice settings from redux state. Part of the overall effort to add store notices to settings and the dashboard. Also renamed `doInitialSetupSuccess` to `settingsBatchRequestSuccess` to be more accurate about what the success action does.

To test:
* `npm run test-client client/extensions/woocommerce` and ensure all tests pass

### CODE REVIEW

File: client/extensions/woocommerce/state/sites/settings/general/selectors.js [hunk 1]
```diff
@@ -111,3 +111,27 @@ export const areTaxCalculationsEnabled = ( state, siteId = getSelectedSiteId( st
...
+	}
+	if ( ! ( 'value' in noticeEnabled ) ) {
+		return false;
+	}
+	return 'yes' === noticeEnabled.value;
+};
+
+export const getStoreNotice = ( state, siteId = getSelectedSiteId( state ) ) => {
+	const generalSettings = getRawGeneralSettings( state, siteId );
+	const notice = find( generalSettings, { id: 'woocommerce_demo_store_notice' } );
```
Comment (psealock) [inline client/extensions/woocommerce/state/sites/settings/general/selectors.js]: naive question here: How come hard coded ids `woocommerce_demo_store_notice` and `woocommerce_demo_store`? Are these simply demo values?

File: client/extensions/woocommerce/state/sites/settings/general/selectors.js [hunk 2]
```diff
@@ -111,3 +111,27 @@ export const areTaxCalculationsEnabled = ( state, siteId = getSelectedSiteId( st
...
+	if ( ! notice ) {
+		return null;
+	}
+	if ( ! ( 'value' in notice ) ) {
```
Comment (psealock) [inline client/extensions/woocommerce/state/sites/settings/general/selectors.js]:
```js
return notice.value || null;
```
Might be shorter? Just a small nit, up to you

File: client/extensions/woocommerce/state/sites/settings/general/selectors.js (hunk 1 above)
Comment (allendav) [reply client/extensions/woocommerce/state/sites/settings/general/selectors.js]: Those are the actual keys from the settings-general REST API elements - their names (i.e. whether this is a demo store or not) aren't quite in line with how they ended up being used over the years (as a means of presenting any notice to customers).

**Comment by allendav** [comment]
@psealock - added unit test for action creator - this PR is fully ready to go

File: client/extensions/woocommerce/state/sites/settings/general/selectors.js (hunk 2 above)
Comment (allendav) [reply client/extensions/woocommerce/state/sites/settings/general/selectors.js]: Sure - makes sense - removed there (and above) in https://github.com/Automattic/wp-calypso/pull/20721/commits/3647e0012fcad2b35f8d9d5c139f5e2d157737d1

**Comment by allendav** [comment]
Re-opening this PR for further work - it turns out that these settings are being removed from WooCommerce and will be managed by the customizer instead.

cc @timmyc

**Comment by timmyc** [comment]
@allendav so based off the latest discussions in #20624 - it looks like we need to know if a store notice is set or not correct? So it seems like this PR would still be applicable since it will allow us to get the `woocommerce_demo_store_notice` value - and `woocommerce_demo_store` value might be of use too in toggling the display of the dashboard widget accordingly too.

Related changes in 3.3:
https://github.com/woocommerce/woocommerce/blob/3.3.0-beta.1/includes/customizer/class-wc-customizer.php#L187-L196

**Pull request closed without merging**